dask/dataframe/io/parquet/core.py

```diff
@@ -97,12 +98,10 @@ def __getitem__(self, key):
             self.fs,
             self.engine.read_partition,
             self.meta,
-            [p["piece"] for p in part],
+            [(p["piece"], p.get("kwargs", {})) for p in part],
             self.columns,
             self.index,
-            toolz.merge(
-                self.common_kwargs, part[0].get("kwargs", {}), self.kwargs or {}
-            ),
+            toolz.merge(self.common_kwargs, self.kwargs or {}),
```

Comment:
Part-specific `kwargs` cannot be merged with the the `common_kwargs` if multiple "pieces" are being accessed for this output partition.